### PR TITLE
Use global date format: %d-%m-%Y

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- https://github.com/the-deep/server/issues/1269

## Changes
- Update visual format to '%d-%m-%Y'

## QA
- Entry Excel export
- Entry Report export
- ARY Export

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: utils/common.py [hunk 1]
```diff
@@ -124,20 +125,22 @@ def get_valid_xml_string(string, escape=True):
     return ''
 
 
-def format_date(date):
-    return date and date.strftime('%d-%m-%Y')
+def deep_date_format(date: Union[datetime.date, datetime.datetime], fallback: Union[str, None] = '') -> Union[str, None]:
```
Comment (bewakes) [inline utils/common.py]: You can use `Optional[str]` which is the prefered shorthand for `Union[str, None]`

**Review by k9845 — APPROVED** [review]
Just curious, why are we using this date format `'%d-%m-%Y'`?

File: utils/common.py (hunk 1 above)
Comment (thenav56) [reply utils/common.py]: Done.

File: utils/common.py [hunk 2]
```diff
@@ -124,20 +125,22 @@ def get_valid_xml_string(string, escape=True):
     return ''
 
 
-def format_date(date):
-    return date and date.strftime('%d-%m-%Y')
+def deep_date_format(date: Union[datetime.date, datetime.datetime], fallback: Optional[str] = '') -> Optional[str]:
+    if date:
```
Comment (bewakes) [inline utils/common.py:132]: This statement implies that the date can be `None`. However the date argument signature says it cannot be `None`. Need to add `None` inside the Union.

**Review by bewakes — CHANGES_REQUESTED** [review]
Looks good to me. A minor request for change.

File: utils/common.py (hunk 2 above)
Comment (thenav56) [reply utils/common.py:132]: Done

**Review by bewakes — APPROVED** [review]

**Review by bimal125 — APPROVED** [review]

**Pull request merged**